src/Tools/_framework/CodeMirror.jsx

```diff
@@ -201,6 +201,11 @@ export default function CodeMirror({
       onBlurExtension,
       onFocusExtension,
       EditorState.changeFilter.of(changeFunc),
+      EditorView.updateListener.of(function (update) {
+        if (update.docChanged) {
+          onBeforeChange(update.state.sliceDoc());
+        }
+      }),
```

Comment:
I assume there are cases that changeFilter runs but not updateListener, so we need both.

Is there a reason you don't use the same changeFunc?  They look to be of the same, except for wrapping with useCallback.

Reply:
I think earlier while I was working on it I assumed the interface for the events handled by updateListener and changeFilter would be different, but after trimming stuff away from examples for updateListener they did end up the same. I'll change it to just pass changeFunc.